### PR TITLE
Structured Binding

### LINKED ISSUE #6724: Structured Binding
### Description

Creating a documentation file on below topic,

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by Megha0606**
<hr>

##  Description 📜

Added the file, Please Review.
Fixes #6724

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [X] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [X] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [X] I have performed a self-review of my own code or work.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generates no new warnings.
- [X] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: C_CPP/Structured Binding/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+// C++17
+auto& [first, second, third, forth] = t;
+
+auto const& [const_first, const_second, const_third, const_forth] = t;
+```
+**Implementation** 
+
+With the introduction of structured bindings, we can now obtain a reference to the tuple members using auto&:
+
+```
```
Comment (apeksha235) [inline C_CPP/Structured Binding/README.md:69]: use ```cpp syntax highlighting for every code

**Review by apeksha235 — CHANGES_REQUESTED** [review]
@Megha0606 also add a relative link of your file in the readme of the respective folder

File: C_CPP/Structured Binding/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+Structured Bindings allow us to define several objects in one go, in a more natural way than in the previous versions of C++.
+
+Consider The Function:
+```
+std::tuple<char, int, bool> mytuple()
+
+{
+
+    char a = 'a';
```
Comment (Suvraneel) [inline C_CPP/Structured Binding/README.md:11]:
```suggestion
std::tuple<char, int, bool> mytuple()  
{  
    char a = 'a';
```

Remove extra lines

File: C_CPP/Structured Binding/README.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    return std::make_tuple(a, i, b);
+}
+```
+
+This returns three variables all of different types. To access these from a calling function prior to C++17, we would need something like:
+
+    char a;
+    int i;
+    bool b;
+    std::tie(a, i, b) = mytuple();
```
Comment (Suvraneel) [inline C_CPP/Structured Binding/README.md:23]:
```suggestion
    ```cpp
    char a;
    int i;
    bool b;
    std::tie(a, i, b) = mytuple();
    ```
```

File: C_CPP/Structured Binding/README.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+# Syntax
+
+    auto ref-opeattr can be any attribute.
+attr can be any attribute.
+
+cv-auto is cv- qualified type specifier auto.
+
+ref-operator is & or &&.
+
+identifier-list is a comma-separated list of variables.
```
Comment (Suvraneel) [inline C_CPP/Structured Binding/README.md:42]:
```suggestion
    **auto ref-operator** can be any attribute.
**attr** can be any attribute.  
**cv-auto** is cv-qualified type specifier auto.  
ref-operator is & or &&.  
**identifier-list** is a comma-separated list of variables.
```

Embolden wherever necessary to make text more legible

**Review by Suvraneel — CHANGES_REQUESTED** [review]
- Syntax Highlighting -> Add `CPP` to each codeblock
- Readme relative link
- Remove unnecessary new line characters...
To add new line you just add two spaces at the end of the previous line

**Comment by Suvraneel** [comment]
@Megha0606  Awesome, Everything else is okay but its not in correct path...
You can NOT create a new folder under our main CPP directory...

Put this under any pre-existing appropriate directory (maybe `C++17` )

**Review by Suvraneel — CHANGES_REQUESTED** [review]
New directory not allowed

**Comment by Suvraneel** [comment]
@Megha0606  LGTM ❤️ :octocat:  🚀  Awesome 💯

**Pull request merged**